projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -411,6 +411,9 @@ export class IgxGridNavigationService {
                 }
 
                 if (this.grid.crudService.cellInEditMode || this.grid.crudService.rowInEditMode) {
+                    // when esc is pressed and editing is canceled, new value is reverted to the old value
+                    const editCellEsc = this.grid.crudService.cell;
+                    editCellEsc.editValue = editCellEsc.value;
                     this.grid.crudService.endEdit(false, event);
```

Comment:
This fix needs to be addressed in the crudService .. there are a lot of operations in the grid that triggered the escape of edit mode for the cell .. /so the cellEditExit event is emitted/ .. you can see the table in the [wiki](https://github.com/IgniteUI/igniteui-angular/wiki/Grid-Editing#common-behaviors)

Reply:
Ok thank you will look into implementing that change